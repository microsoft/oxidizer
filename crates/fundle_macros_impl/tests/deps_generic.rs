// Copyright (c) Microsoft Corporation.

use syn::{parse_quote,  ItemStruct};

mod util;

#[test]
fn lifetimes() {
    let item: ItemStruct = parse_quote! {
        #[deps]
        struct Foo<'a> {
            x: &'a u32
        }
    };

    insta::assert_snapshot!(expand_fundle_deps!(item));
}

#[test]
fn generics() {
    let item: ItemStruct = parse_quote! {
        #[deps]
        struct Foo<T> {
            x: T
        }
    };

    insta::assert_snapshot!(expand_fundle_deps!(item));
}

#[test]
fn generics_where() {
    let item: ItemStruct = parse_quote! {
        #[deps]
        struct Foo<T> where T: Clone {
            x: T
        }
    };

    insta::assert_snapshot!(expand_fundle_deps!(item));
}