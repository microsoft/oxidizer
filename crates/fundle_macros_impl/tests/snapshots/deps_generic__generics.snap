---
source: crates/fundle_macros_impl/tests/deps_generic.rs
expression: expand_fundle_deps!(item)
---
#[allow(private_bounds)]
struct Foo<T> {
    x: T,
}
#[allow(private_bounds)]
impl<T, __FundleFromT> ::std::convert::From<__FundleFromT> for Foo<T>
where
    __FundleFromT: ::std::convert::AsRef<T>,
{
    fn from(value: __FundleFromT) -> Self {
        Self {
            x: <__FundleFromT as ::std::convert::AsRef<T>>::as_ref(&value).to_owned(),
        }
    }
}
